smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/readwrite/ServiceUtils.kt

```diff
@@ -0,0 +1,54 @@
+package software.amazon.smithy.swift.codegen.integration.serde.readwrite
+
+import software.amazon.smithy.aws.traits.protocols.AwsJson1_0Trait
+import software.amazon.smithy.aws.traits.protocols.AwsJson1_1Trait
+import software.amazon.smithy.aws.traits.protocols.AwsQueryTrait
+import software.amazon.smithy.aws.traits.protocols.Ec2QueryTrait
+import software.amazon.smithy.aws.traits.protocols.RestJson1Trait
+import software.amazon.smithy.aws.traits.protocols.RestXmlTrait
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.swift.codegen.model.hasTrait
+
+// An enum expressing the six defined AWS protocols used with Smithy.
+enum class AWSProtocol {
```

Comment:
This file provides methods and types for easily determining & expressing the wire protocols of a service.